simulators/main/simulators.py

```diff
@@ -61,6 +63,13 @@ def tick(self):
             return
         if self.machine.should_tick():
             for task in self.machine.tasks:
+                try:
+                    if self.machine.first_time:
+                        # set the next_run time to always let the measurement generation to run in the first time
+                        task.next_run = datetime.timestamp(datetime.utcnow()) + 1
```

Comment:
does that add 1 millisecond? I'm not sure, but could this cause problems with the timing during initialization? Maybe we should at least add 1...5 sesonds?

Reply:
Since the tick() function runs immediately after this line and the datetime.utcnow() is also used to check the current time, I think one milliseconds is enough. If the additional time is much higher, there might be some distortions with the timestamps